api/auth.proto

```diff
@@ -44,3 +46,78 @@ message AuthAction {
   // List of rules
   repeated Rule rule = 2;
 }
+
+service Authorization {
+	rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message CheckRequest {
+
+	// Address of the downstream source of the request.
+	Address source = 1;
+
+	// Address of the upstream destination of the request.
+	Address destination = 2; 
```

Comment:
I don't think in any of the cases where a filter is running we are going to know this without major changes. I would remove this for now.

Reply:
Not even the original destination port/address?  I.e. before routing/load balancing etc.?

Reply:
Yes we know that. I would rename source -> downstream_remote_address, and destination -> downstream_local_address

Reply:
remote v. local assumes authorization is used for ingress.  Although that's our first use case, we will eventually want to consider egress authorization.  Is there a term that would make sense in either arrangement, like original_destination?

Reply:
I'm not sure I follow. What data are you trying to send the auth server?

Reply:
If my local workload tries to make a connection to a remote network location (perhaps outside the service mesh), I want have the auth server be able to see where they were trying to connect to before I allow it.

In HTTP, I can probably tell from the host header, but for other protocols I may need to depend on network layer information like the IP & port.

Reply:
downstream_local_address would be the local IP/port that the client connected to in Envoy. The only other data would be the "original destination" if using IP tables and original destination routing. Is that what you are after?

Reply:
Yeah, in most circumstances we'd be interested in what the destination was prior to any iptables redirects to Envoy, although it might be useful to pass both so that we can deal with different use cases.